mqtt/tracing/handler.go

```diff
@@ -0,0 +1,90 @@
+package tracing
+
+import (
+	"context"
+
+	"github.com/mainflux/mproxy/pkg/session"
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	authConnectOP   = "auth_connect_op"
+	authPublishOP   = "auth_publish_op"
+	authSubscribeOP = "auth_subscribe_op"
+	connectOP       = "connect_op"
+	disconnectOP    = "disconnect_op"
+	subscribeOP     = "subscribe_op"
+	unsubscribeOP   = "unsubscribe_op"
+	publishOP       = "publish_op"
+)
+
+var _ session.Handler = (*handlerMiddleware)(nil)
+
+type handlerMiddleware struct {
+	handler session.Handler
+	tracer  opentracing.Tracer
+}
+
+// NewHandler creates a new session.Handler middlware with tracing.
+func NewHandler(tracer opentracing.Tracer, handler session.Handler) session.Handler {
+	return &handlerMiddleware{
+		tracer:  tracer,
+		handler: handler,
+	}
+}
+
+// AuthConnect traces auth connect operations.
+func (h *handlerMiddleware) AuthConnect(ctx context.Context) error {
+	span, ctx := opentracing.StartSpanFromContextWithTracer(ctx, h.tracer, authConnectOP)
+	defer span.Finish()
+	return h.handler.AuthConnect(ctx)
+}
+
+// AuthPublish traces auth publish operations.
+func (h *handlerMiddleware) AuthPublish(ctx context.Context, topic *string, payload *[]byte) error {
+	span, ctx := opentracing.StartSpanFromContextWithTracer(ctx, h.tracer, authPublishOP)
+	defer span.Finish()
+	return h.handler.AuthPublish(ctx, topic, payload)
```

Comment:
Should we enrich the context with the topic and, maybe, publisher ID?  The same goes for other operations.